core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -510,12 +516,19 @@ private[zio] final class FiberContext[E, A](
                 case ZIO.Tags.FiberRefModify =>
                   val zio = curZio.asInstanceOf[ZIO.FiberRefModify[Any, Any]]
 
-                  val oldValue           = Option(fiberRefLocals.get(zio.fiberRef))
+                  val oldValue = FiberRef
+                    .popPendingUnsafeSet(zio.fiberRef)
+                    .orElse(Option(fiberRefLocals.get(zio.fiberRef)))
+
                   val (result, newValue) = zio.f(oldValue.getOrElse(zio.fiberRef.initial))
                   fiberRefLocals.put(zio.fiberRef, newValue)
 
                   curZio = nextInstr(result)
 
+                case ZIO.Tags.SetCurrentFiberContext =>
+                  FiberContext.current.set(this)
```

Comment:
Why do we need this, since if `fiberContextPropagation` is set, then `FiberContext.current.set(this)` is run at the start of the method `evaluateNow`, and of course will not change until the end.

Reply:
Indeed, this is just a leftover from an older implementation that tried to be "smart". I removed it.